netbox/netbox/forms/base.py

```diff
@@ -39,7 +39,7 @@ def _get_form_field(self, customfield):
             form_field = customfield.to_form_field(set_initial=False)
             initial = self.instance.custom_field_data.get(customfield.name)
             if customfield.type == CustomFieldTypeChoices.TYPE_JSON:
-                form_field.initial = json.dumps(initial)
+                form_field.initial = json.dumps(initial) if initial else ''
```

Comment:
I think we need to check against None explicitly, because e.g. `0` would be a valid value.
```suggestion
                form_field.initial = json.dumps(initial) if initial is not None else ''
```

Reply:
But that wouldn't be valid JSON, I think you have to have key:value for valid JSON which would never evaluate to False?

Reply:
AFAIK you can store arbitrary scalars like `123` or `"foo"` in the field too.

Reply:
Hi all, I am the reported of the original issue.  This PR fixes the issue when the field is null, but existing data is still ending up escaped.